File: python/Examples/09_waveform_streaming_test.py
from HelpFunctions.Engine import EA_Engine
import HelpFunctions.ea_engine as EA

# Runs a waveform streaming test which lets the user choose the signal
if __name__ == "__main__":
    engine = EA_Engine()
    engine.deselectWdmInputDevice()
    engine.setInputChannel(1, "4189", True)
    engine.setOutputChannel(1, "3670-A", True)
    path = "wav_for_streaming.wav"
    # The results from the test will be saved in the res variable. If the plots are active you have to close them before the function returns
    res = engine.runWaveformStreamingTest(1, path, trackIndex=1)

File: csharp/csharp/Examples/09_waveform_streaming_test.cs
namespace EAProject.Examples
{
    // Runs a waveform streaming test which lets the user choose the signal
    class Waveform_streaming_test
    {
        public static void Main()
        {
            var engine = new EAProject.EA_Engine();
            engine.DeselectWdmInputDevice();
            engine.SetAllChannelsToFalse();
            engine.SetInputChannel(1, "4189 Ch.1", true);
            engine.SetOutputChannel(1, "3670-A", true);
            string fileName = "wav_for_streaming.wav";
            string projectDirectory = AppDomain.CurrentDomain.BaseDirectory;
            string parentDirectory = Directory.GetParent(projectDirectory).Parent.Parent.Parent.FullName;
            string path = Path.Combine(parentDirectory, fileName);
            // The results from the test will be saved in the res variable. If the plots are active you have to close them before the function returns
            var res = engine.RunWaveformStreamingTest(1, path, trackIndex: 1);
        }
    }
}
